src/librustc_borrowck/diagnostics.rs

```diff
@@ -429,6 +429,93 @@ You can find more information about borrowing in the rust-book:
 http://doc.rust-lang.org/stable/book/references-and-borrowing.html
 "##,
 
+E0509: r##"
+This error occurs when an attempt is made to move out of a container that
```

Comment:
"...a value whose type implements..." sounds kind of awkward to me, but I agree it's less ambiguous.